src/WaitForContainer.js

```diff
@@ -0,0 +1,65 @@
+import PropTypes from 'prop-types'
+import componentOrElement from 'prop-types-extra/lib/componentOrElement'
+import canUseDom from 'dom-helpers/util/inDOM'
+import ownerDocument from 'dom-helpers/ownerDocument'
+import React from 'react'
+import ReactDOM from 'react-dom'
+import getContainer from './utils/getContainer'
+
+const propTypes = {
+  /**
+   * A Node, Component instance, or function that returns either. The `container` will have the Portal children
+   * appended to it.
+   */
+  container: PropTypes.oneOfType([componentOrElement, PropTypes.func]),
+
+  onContainerResolved: PropTypes.func,
+}
+
+class WaitForContainer extends React.Component {
+  constructor(...args) {
+    super(...args)
+
+    if (!canUseDom) return
+
+    let { container } = this.props
+
+    if (typeof container === 'function') container = container()
+    if (container && !ReactDOM.findDOMNode(container)) {
```

Comment:
O_O i'm surprised this is safe

Reply:
Yeah me too...I had it wrapped in a try/catch but no errors were ever thrown so idk